sentry_sdk/integrations/asgi.py

```diff
@@ -213,6 +200,46 @@ def event_processor(self, event, hint, asgi_scope):
     # data to your liking it's recommended to use the `before_send` callback
     # for that.
 
+    def _set_transaction_name_and_source(self, event, transaction_style, asgi_scope):
```

Comment:
I just wanted to make it extra clear that the event is manipulated. (and also make it a bit looking like functional programming) But I guess it is python, so you are right :-)

Reply:
done

Reply:
so as a thumbrule in dynamic languages, I abide by the following pattern.
If I take some input and am returning something, I try to make the function as immutable as possible, so if I'm returning the same object it needs to be copied and not mutated.
If the function returns nothing anyway, then it's clear it does something which has side effects and then it can mutate some of it's inputs.